File: app/src/main/java/mb/pokequiz/data/model/Pokedex.java
package mb.pokequiz.data.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by mbpeele on 2/1/16.
 */
public class Pokedex {

    private String created;
    private String modified;
    private String name;
    @SerializedName(SerializedNames.POKEDEX_ITEMS) private List<DexItem> dexItems;
    @SerializedName(SerializedNames.RESOURCE_URI) private String uri;

    public String getUri() {
        return uri;
    }

    public void setUri(String resource_uri) {
        this.uri = resource_uri;
    }

    public List<DexItem> getDexItems() {
        return dexItems;
    }

    public void setPokemon(List<DexItem> pokemon) {
        this.dexItems = pokemon;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getModified() {
        return modified;
    }

    public void setModified(String modified) {
        this.modified = modified;
    }

    public String getCreated() {
        return created;
    }

    public void setCreated(String created) {
        this.created = created;
    }

    public static class DexItem {

        private String name;
        @SerializedName(SerializedNames.RESOURCE_URI) private String uri;

        public String getUri() {
            return uri;
        }

        public void setUri(String resource_uri) {
            this.uri = resource_uri;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }
}

File: app/src/main/java/mb/pokequiz/web/model/Pokedex.kt
package mb.pokequiz.web.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Pokedex {

    /**

     * @return
     * *     The id
     */
    /**

     * @param id
     * *     The id
     */
    @SerializedName("id")
    @Expose
    var id: Int? = null
    /**

     * @return
     * *     The name
     */
    /**

     * @param name
     * *     The name
     */
    @SerializedName("name")
    @Expose
    var name: String? = null
    /**

     * @return
     * *     The isMainSeries
     */
    /**

     * @param isMainSeries
     * *     The is_main_series
     */
    @SerializedName("is_main_series")
    @Expose
    var isMainSeries: Boolean? = null
    /**

     * @return
     * *     The descriptions
     */
    /**

     * @param descriptions
     * *     The descriptions
     */
    @SerializedName("descriptions")
    @Expose
    var descriptions: List<Description>? = null
    /**

     * @return
     * *     The names
     */
    /**

     * @param names
     * *     The names
     */
    @SerializedName("names")
    @Expose
    var names: List<Name>? = null
    /**

     * @return
     * *     The pokemonEntries
     */
    /**

     * @param pokemonEntries
     * *     The pokemon_entries
     */
    @SerializedName("pokemon_entries")
    @Expose
    var pokemonEntries: List<PokemonEntry>? = null
    /**

     * @return
     * *     The region
     */
    /**

     * @param region
     * *     The region
     */
    @SerializedName("region")
    @Expose
    var region: Region? = null
    /**

     * @return
     * *     The versionGroups
     */
    /**

     * @param versionGroups
     * *     The version_groups
     */
    @SerializedName("version_groups")
    @Expose
    var versionGroups: List<VersionGroup>? = null

}
